README.md

```diff
@@ -1,29 +1,91 @@
 
-## Contributing
-All contributions are welcome: ideas, patches, documentation, bug reports, complaints, and even something you drew up on a napkin.
-
-## Developing
-To better understand contributing to logstash plugins, visit [their article on custom output plugins](https://www.elastic.co/guide/en/logstash/current/_how_to_write_a_logstash_output_plugin.html).
+# Installation
+### Install from Ruby Gem
+```
+gem install logstash-output-wavefront
+```
 
-To test a local build of this plugin,
+### Install from Source
+To Install from source follow the below steps:
+   1. Install ruby.
+   2. Install ruby bundler -- `gem install bundler`.
+   3. Clone this repository and `cd` to the directory.
+   4. Build the plugin -- `gem build logstash-output-wavefront.gemspec`.
+   5. Install the plugin --`logstash-plugin install *wavefront*.gem`.
 
-   1. Install logstash. Either from source or from a package manager is fine.
-   1. Install ruby bundler -- `gem install bundler` should do it.
-   1. In this directory, `gem build logstash-output-wavefront.gemspec`
-   1. `logstash-plugin install *wavefront*.gem`
+# Enable Output plugin
+Create/Update a config file that specifies `wavefront` as output plugin
+```
+output {
+    wavefront {
+      host => "<Proxy-IP>"
+    }
+}
+```
+#### Optional Configuration
+```
+  port          Metric Port (Default - 2878)
+  prefix        Metric Prefix (Default - "logstash")
+  metrics       List of metrics (Default - ["count", "mean"])
+  source        Metric source (Default - Hostname of the node running logstash)  
+```
 
-### Install dependencies
+# Log Data Format
+Wavefront output plugin for Logstash can only process the log data as metric which is in below format
+```
+{
+   "bytes" => {
+     "count" => 200,
+     "mean" => 42.2
+   },
+   "error" => {
+     "count" => 123,
+     "code" => 404
+   },
+   "message" => "I'm not a hash type, so I won't get sent."
+ }
 ```
-bundle install
+Wavefront output plugin for Logstash generate below metrics out of the above log data and sends to Wavefront
 ```
+logstash.bytes.count 200
+logstash.bytes.mean 42.2
+logstash.error.count 123
+```
+Wavefront output plugin for Logstash also supports sending point tags for a metric, below is the log data format
+```
+{
+   "bytes.tagz.type=access.region=mumbai" => {
+     "count" => 200,
+     "mean" => 42.2
+   },
+   "error" => {
+     "count" => 123,
+     "code" => 404
+   },
+   "message" => "I'm not a hash type, so I won't get sent."
+ }
+```
+Below metrics are the output of the above log data
+```
+logstash.bytes.count 200 type=access region=mumbai
+logstash.bytes.mean 42.2 type=access region=mumbai
+logstash.error.count 123
+```
+
+**Note:** Wavefront output plugin for Logstash has dropped out the `logstash.error.code` metric, as default `metrics` list only includes `count and mean`, to include the `code` metric you have to override the default `metrics` to `["count", "mean", "code"]`.
```

Comment:
Where is the default metric list and how can user change this?

Reply:
Default metric list I have mentioned in the Optional Configuration section of the Wavefront output plugin.